### PR TITLE
Home Fishtank

### PR DESCRIPTION
**Pull request by imgntn**
This PR adds a fishtank for the Home space.

To test:
1. Run this script https://rawgit.com/imgntn/hifi/newfishes/unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js
2. Observe the fish swimming around inside of the tank
3. They will swim toward where you're looking when you're close to the tank.

(Note: cleanup is on by default right now -- when this is integrated into some kind of home spawner/reset script, it will persist)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: fdaa1f9ded4ce70d1b55d8a0c5bfe6994a696bb6 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 249d88b619aa1bf26ac33bda3e15990e7e0a60ae available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7

Couple things.  Fish seem to get stuck and oscillate at times (I can upload a video if that's not something that makes sense for description) and - nitpicking yes - but real fish hardly, if ever, swim 100% vertical straight line. ;)  Beyond nitpicking this 99% passes QA, no stuck fish freaking out would be 100% Pass.

**Comment by imgntn** [comment]
@OmegaHeron ha, thanks, yes -- the swimming algorithm could use some tuning for a few more parameters (i.e. don't pitch more than 30 degrees in either direction), but i think we're going to get this version of the code in and then incrementally improve from there.  thanks for the notes!

**Comment by OmegaHeron** [comment]
Yes, it seems more than good enough to get into codebase and improve upon.

File: unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    bubbleSystem = Entities.addEntity(bubbleProperties);
+    createBubbleSound(finalOffset);
+}
+
+function createSecondBubbleSystem() {
+
+    var tankProperties = Entities.getEntityProperties(fishTank);
+    var bubbleProperties = {
+        "name": 'hifi-home-fishtank-bubbles',
+        "color": {},
```
Comment (jherico) [inline unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js]: que?

File: unpublishedScripts/DomainContent/Home/fishTank/tank.js [hunk 2]
```diff
@@ -0,0 +1,696 @@
+(function() {
+
+    Script.include('../../../../examples/libraries/virtualBaton.js?' + Math.random());
```
Comment (jherico) [inline unpublishedScripts/DomainContent/Home/fishTank/tank.js]: shouldn't include code to defeat the cache in the production version.

File: unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js (hunk 1 above)
Comment (imgntn) [inline unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js]: that's what the particle explorer exports -- i guess i can remove the unused colors

File: unpublishedScripts/DomainContent/Home/fishTank/tank.js (hunk 2 above)
Comment (imgntn) [inline unpublishedScripts/DomainContent/Home/fishTank/tank.js]: :+1:

**Comment by hifi-gustavo** [comment]
Build console output for commit: 645f0da280f9e5f2ef96d2fe169cb829f44af2b7 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4fd9d4157b5e42fe80d5a8f2b6c36f3f4a482c6 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

File: unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+Script.setTimeout(function() {
+    print('CREATE TIMEOUT!!!')
+    print('TANK AT CREATE IS::: ' + fishTank)
+    print('DATA AT CREATE IS:::' + JSON.stringify(data));
+
+    setEntityCustomData(customKey, fishTank, data);
+    // setEntityCustomData('grabbableKey', id, {
+    //     grabbable: false
+    // });
+}, 2000)
```
Comment (sethalves) [inline unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js]: Should these prints or commented code be removed?

File: unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js [hunk 4]
```diff
@@ -0,0 +1,422 @@
+var fishTank, tankBase, bubbleSystem, secondBubbleSystem, innerContainer, bubbleInjector, lowerCorner, upperCorner, urchin, rocks;
```
Comment (sethalves) [inline unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js]: The copyright blob fell a ways down...

File: unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+    Script.scriptEnding.connect(cleanup);
+}
+
+//  Copyright 2016 High Fidelity, Inc.
+//
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
```
Comment (sethalves) [inline unpublishedScripts/DomainContent/Home/fishTank/createFishTank.js]: move copyright back to top of file

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3a5f7f1cbed4a3345a4103ff191f9c9d98459377 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a8ac556103c1bc05f3f9e369df3d82fa2160b9f4 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c05e996913a8c2b829b26bc26e9df39cfdaa1a66 available at: https://gist.github.com/dceb50bad31089daf42c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7261.dmg

**Comment by ericrius1** [comment]
Besides the occasional flickering and fishes caught on edge of tank, this works really well. Perhaps with V2 we can figure out issues, but is definitely good enough to merge

**Pull request merged**